### PR TITLE
Add corrector class and add validate_name method

### PR DESCRIPTION
**Pull request by JbirdL86**
- Create a class Corrector with a method correct_name 
  - Make sure the first letter of the given name is a capital letter.
  - Make sure that the name has a maximum of 10 characters. If it's longer it should trim the word.

- Set an instance of Corrector in Person on initialization.
  - Add a method validate_name that uses the Corrector instance to validate and save the corrected name

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Juan , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

The linters for your project are failing, please make sure all linters are passing before requesting another code review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: corrector.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Corrector
+  def correct_name(name)
+    name.slice(0, 10).capitalize
```
Comment (o-rashad) [inline corrector.rb]: Great work on creating the `Corrector` class with `correct_name` method... 💯 
only one thing i want to mention is 

 - `capitalize` returns a new string with the first letter capitalized
 - `capitalize!` modifies or mutates the original string 

For further read [LINK](https://stackoverflow.com/questions/26103869/using-capitalize-or-capitalize)


- [ ] Change `capitalize` to `capitalize!` to mutate the user input name to capitalize it if it's not. 👍

**Review by o-rashad — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Asslam Alikom @JbirdL86  👋😄,
I'm Omar and I will be your reviewer for this project 😊

Good job so far!👌😃

There are some issues 🐞 that you still need to work on to go to the next project but you are almost there! 🏁🏁🏁

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy coding!👏:heart: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: corrector.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Corrector
```
Comment (SajjadAhmad14) [inline corrector.rb:1]: [  OPTIONAL ] Please make sure all the linter checks are passed on PR.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Juan, 
## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏
# OPTIONAL BUT RECOMMENDED
- [ ] Please see inline comments.
## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**